### PR TITLE
Feat/prod3 89

### PR DESCRIPTION
**Pull request by nathielymm**
# Description

- Adds recurrence page on payment form

## Does this pull request close any open issues?

-  closes [PROD3-87](https://ribon.atlassian.net/browse/PROD3-87)
- closes [PROD3-89](https://ribon.atlassian.net/browse/PROD3-89)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Dangerous changes
- [ ] Chore (change that does not effect how the code works, eg: change color)
- [ ] Tests

## How to test

Put here all the things need to test this PR and verify your changes, also list any relevant details for tests (eg: have a wallet)

- Test A

### CODE REVIEW

**Review by thaisribon — COMMENTED** [review]
the "alterar" button has a problem
WhatsApp Image 2023-08-14 at 16 25 25

File: src/screens/promoters/RecurrenceScreen/index.tsx [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+  } = params;
+
+  useEffect(() => {
+    if (targetIdParam) setTargetId(targetIdParam);
+    if (targetParam) setTarget(targetParam);
+    if (currencyParam) setCurrency(currencyParam);
+    if (offerParam != null) {
+      setOffer(offerParam);
+    }
+  }, [targetParam, currencyParam, targetParam, offerParam, offers]);
```
Comment (thaisribon) [inline src/screens/promoters/RecurrenceScreen/index.tsx:71]: here, I think we could separate in different useEffects, what do you think?

**Comment by nathielymm** [comment]
> the "alterar" button has a problem WhatsApp Image 2023-08-14 at 16 25 25

Fixed it

File: src/screens/promoters/RecurrenceScreen/index.tsx [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    keyPrefix: "promoters.recurrenceScreen",
+  });
+
+  const [currentOffer, setCurrentOffer] = useState<Offer>();
+  const [currentIndex, setCurrentIndex] = useState<number>();
+
+  const [offersModalVisible, setOffersModalVisible] = useState(false);
+
+  const {
+    target,
```
Comment (thaisribon) [inline src/screens/promoters/RecurrenceScreen/index.tsx:35]: I think this might be the problem, because we declare a const named target below

**Review by thaisribon — APPROVED** [review]

**Pull request merged**